specification/azuredata/resource-manager/Microsoft.AzureData/preview/2019-07-24-preview/azuredata.json

```diff
@@ -1984,6 +2272,97 @@
         }
       }
     },
+    "SqlServerInstanceProperties": {
+      "description": "Properties of SqlServerInstance.",
+      "type": "object",
+      "properties": {
+        "version": {
+          "type": "string",
+          "description": "SQL Server version."
+        },
+        "edition": {
+          "type": "string",
+          "description": "SQL Server edition."
+        },
+        "containerResourceId": {
+          "type": "string",
+          "description": "ARM Resource id of the container resource (Azure Arc for Servers)"
+        },
+        "createTime": {
+          "type": "string",
+          "description": "The time when the resource was created."
+        },
+        "updateTime": {
+          "type": "string",
+          "description": "The time when the resource was last updated."
+        },
+        "vCore": {
+          "type": "string",
+          "description": "The number of logical processors used by the SQL Server instance."
+        },
+        "status": {
+          "type": "string",
+          "description": "The cloud connectivity status."
+        }
+      },
+      "required": [
+        "version",
+        "edition",
+        "containerResourceId",
+        "createTime",
+        "updateTime",
```

Comment:
createTime, updateTime, and status aren't actually required to be entered by the user right?  Wouldn't they be readonly, generated by the RP?

Reply:
"status" needs to be input by user because we are not sure if an auto detection will be added at this moment. But we may re-consider it later.
For "createTime" and "updateTime", you are right that it should be auto generated. The code is not in place yet for the private preview timeline.

Reply:
createTime and updateTime should still be removed from the required list and marked readonly.  The user should not b entering these even if you aren't populating them in the backend yet

Reply:
will remove createTime and updateTime from the required list.

Reply:
createTime and updateTime are set to readonly now.